lib/ansible/modules/cloud/pubnub/pubnub_blocks.py

```diff
@@ -0,0 +1,2090 @@
+#!/usr/bin/python
+#
+# PubNub Real-time Cloud-Hosted Push API and Push Notification Client Frameworks
+# Copyright (C) 2016 PubNub Inc.
+# http://www.pubnub.com/
+# http://www.pubnub.com/terms
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'commiter',
+                    'version': '1.0'}
+
+
+DOCUMENTATION = '''
+---
+module: pubnub_blocks
+version_added: '2.2'
+short_description: PubNub blocks management module.
+description:
+  - This module allow to manage existing blocks by managing event handlers and starting/stopping block.
+author:
+  - PubNub <support@pubnub.com> (@pubnub)
+  - Sergey Mamontov <sergey@pubnub.com> (@parfeon)
+requirements:
+  - "python >= 2.7"
+options:
+  email:
+    description:
+      - Email from account for which new session should be started.
+      - Not required if C(cache) contains result of previous module call (in same play).
+    required: false
+  pwd:
+    description:
+      - Password which match to account to which specified C(email) belong.
+      - Not required if C(cache) contains result of previous module call (in same play).
+    required: false
+  cache:
+    description:
+      - In case if single play use blocks management module few times it is preferred to enabled 'caching' by making previous module to share gathered artifacts and pass them to this parameter.
+    required: false
+    default: {}
+  application:
+    description:
+      - Name of target PubNub application for which blocks configuration on specific C(keyset) will be done.
+    required: true
+  keyset:
+    description:
+      - Name of application's keys set which is bound to managed blocks.
+    required: true
+  state:
+    description:
+      - Intended block state after event handlers creation / update process will be completed.
+    required: false
+    default: 'start'
+    choices: ['start', 'stop', 'present', 'absent']
+  block:
+    description:
+      - Name of managed block which will be later visible on admin.pubnub.com.
+    required: true
+  description:
+    description:
+      - Short block description which will be later visible on admin.pubnub.com. Used only if block doesn't exists and won't change description for existing block.
+    required: false
+    default: 'New block'
+  event_handlers:
```

Comment:
@ryansb to call pubnub block user need to specify authorization credentials or result of previous call of module inside of single play.  
To manage blocks or their event handlers it is required to specify their name, block doesn't rely on order of items in `event_handlers`.  
If user explicitly doesn't request any changes to concrete event handler, nothing will be done. If some event handler for same block in single play will appear twice and both with `state` set to **present**, module will check whether there is difference in how event handler represented in block at this moment and in playbook and send updates if required.